content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/wait_for_completion.rb

```diff
@@ -43,15 +42,20 @@ def check_status(job)
 
                 def ansible_job
                   job_id = @handle.get_state_var(:ansible_job_id)
+                  job_class = @handle.get_state_var(:ansible_job_class)
                   if job_id.nil?
                     @handle.log(:error, 'Ansible job id not found')
                     exit(MIQ_ERROR)
                   end
-                  fetch_job(job_id)
+                  if job_class.nil?
+                    @handle.log(:error, 'Ansible job class not found')
+                    exit(MIQ_ERROR)
```

Comment:
@lfu can we use a raise here instead of an exit.
If we are going to use exit we need the specs to account for that exit call similar to
https://github.com/ManageIQ/manageiq-content/blob/3f8a65838dc2fdaeb90fa1f5f225e2107749e2eb/spec/content/automate/ManageIQ/System/CommonMethods/Utils.class/__methods__/log_object_spec.rb#L79